packages/headless/src/features/product-recommendations/product-recommendations-analytics.actions.ts

```diff
@@ -13,3 +25,95 @@ export const logProductRecommendations = makeAnalyticsAction(
       state as StateNeededByProductRecommendationsAnalyticsProvider
     )
 );
+
+const partialRecommendationInformation = (
+  result: ProductRecommendation,
+  state: Partial<ProductRecommendationsAppState>
+): PartialDocumentInformation => {
+  const resultIndex =
+    state.productRecommendations?.recommendations.findIndex(
+      ({permanentid}) => result.permanentid === permanentid
+    ) || 0;
```

Comment:
Should we return `null` here if not found ? And then not log a UA event when that happens ?
With a console warning.

Instead of silently failing/fall backing on a "random" result.

Reply:
The findIndex() is only to get the position of the item in the list, to know if the recommendation that was clicked on is the first, second , third...  If not found, the UA event is still valuable even if we default the position to 0.